zingo-netutils/src/lib.rs

```diff
@@ -84,43 +101,51 @@ impl GrpcConnector {
                     })
                     .service(http_connector);
                 let client = Box::new(hyper::Client::builder().build(connector));
-                let uri = uri.clone();
                 let svc = tower::ServiceBuilder::new()
                     //Here, we take all the pieces of our uri, and add in the path from the Requests's uri
-                    .map_request(move |mut req: http::Request<tonic::body::BoxBody>| {
+                    .map_request(move |mut request: http::Request<tonic::body::BoxBody>| {
+                        let path_and_query = request
+                            .uri()
+                            .path_and_query()
+                            .cloned()
+                            .unwrap_or(PathAndQuery::from_static("/"));
                         let uri = Uri::builder()
-                            .scheme(uri.scheme().unwrap().clone())
-                            .authority(uri.authority().unwrap().clone())
+                            .scheme(scheme.clone())
+                            .authority(authority.clone())
                             //here. The Request's uri contains the path to the GRPC sever and
                             //the method being called
-                            .path_and_query(req.uri().path_and_query().unwrap().clone())
+                            .path_and_query(path_and_query)
                             .build()
                             .unwrap();
 
-                        *req.uri_mut() = uri;
-                        req
+                        *request.uri_mut() = uri;
+                        request
                     })
                     .service(client);
 
                 Ok(CompactTxStreamerClient::new(svc.boxed_clone()))
             } else {
                 let connector = tower::ServiceBuilder::new().service(http_connector);
                 let client = Box::new(hyper::Client::builder().http2_only(true).build(connector));
-                let uri = uri.clone();
                 let svc = tower::ServiceBuilder::new()
                     //Here, we take all the pieces of our uri, and add in the path from the Requests's uri
-                    .map_request(move |mut req: http::Request<tonic::body::BoxBody>| {
+                    .map_request(move |mut request: http::Request<tonic::body::BoxBody>| {
+                        let path_and_query = request
+                            .uri()
+                            .path_and_query()
+                            .cloned()
+                            .unwrap_or(PathAndQuery::from_static("/"));
```

Comment:
Do we really want this default?

Reply:
Not really, but map_request is infallible, and I don't know that there's an alternative.

Reply:
So..  don't use map_request?   What's correct behavior here?   Can I trick a client into connecting to the wrong resource?   Is this worse than a panic!?

Reply:
But..   now that's an explicit error variant and return that, instead of panic?